Sources/Shared/DiffAware.swift

```diff
@@ -0,0 +1,47 @@
+//
+//  DiffAware.swift
+//  DeepDiff
+//
+//  Created by khoa on 22/02/2019.
+//  Copyright © 2019 Khoa Pham. All rights reserved.
+//
+
+import Foundation
+
+/// Model must conform to DiffAware for diffing to work properly
+/// idProviding: To tell if there is deletion or insertion
+/// comparing: To tell if there is replacement or movement
+public protocol DiffAware {
+  var idProviding: Int { get }
+  static func comparing(_ a: Self, _ b: Self) -> Bool
```

Comment:
I wonder why you went with a static func instead of a `compareTo` instance func? Not that it makes much difference but was curious

Reply:
It's just my personal taste, I think it's better to use static function when it comes to performing actions on 2 objects